contrib/src/main/java/com/datatorrent/contrib/avro/PojoToAvro.java

```diff
@@ -0,0 +1,296 @@
+      try (ObjectOutputStream o = new ObjectOutputStream(b)) {
+        o.writeObject(obj);
+      }
+      return b.toByteArray();
+    }
+  }
+
+  /**
+   * @param -
+   *          className
+   * @param -
+   *          name of the field to create the getter for Returns a getter
+   */
+
+  private Getter<?, ?> generateGettersForField(Class<?> cls, String inputFieldName)
+      throws NoSuchFieldException, SecurityException
+  {
+    java.lang.reflect.Field f = cls.getDeclaredField(inputFieldName);
+    Class<?> c = ClassUtils.primitiveToWrapper(f.getType());
+
+    Getter<?, ?> classGetter = PojoUtils.createGetter(cls, inputFieldName, c);
+
+    return classGetter;
+  }
+
+  /**
+   * @param -
+   *          schema of the generic record Assumption is that the name of a
+   *          field in POJO is the same as the name in Avro schema
+   */
+
+  public void initializeColumnMap(Schema schema)
+  {
+    columnNames = schema.getFields();
+
+    keyMethodMap = new ArrayList<Getter>();
+    for (int i = 0; i < columnNames.size(); i++) {
+      try {
+        keyMethodMap.add(generateGettersForField(cls, columnNames.get(i).name()));
+      } catch (NoSuchFieldException | SecurityException e) {
+        throw new RuntimeException("Failed to initialize pojo class getters for field: ", e);
+      }
+    }
+  }
+
+  @InputPortFieldAnnotation(optional = true, schemaRequired = true)
+  public final transient DefaultInputPort<Object> data = new DefaultInputPort<Object>()
+  {
+
+    @Override
+    public void setup(PortContext context)
+    {
+      cls = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+
+    @Override
+    public void process(Object tuple)
+    {
+
+      GenericRecord record = null;
+
+      try {
+        record = getGenericRecord(tuple);
+      } catch (Exception e) {
+        LOG.error("Exception in parsing record");
+        errorCnt++;
+      }
+
+      if (record != null) {
+        output.emit(record);
+        recordCnt++;
+      } else {
+        errorCnt++;
+      }
+
+    }
+
+  };
+
+  /**
+   * @param -Object
+   *          Returns a generic record mapping the POJO fields to provided
+   *          schema
+   */
+
+  public GenericRecord getGenericRecord(Object tuple) throws Exception
+  {
+
+    GenericRecord rec = new GenericData.Record(getSchema());
+
+    for (int i = 0; i < columnNames.size(); i++) {
+      try {
+        rec.put(columnNames.get(i).name(), AvroRecordHelper.convertValueStringToAvroKeyType(getSchema(),
+            columnNames.get(i).name(), keyMethodMap.get(i).get(tuple).toString()));
+      } catch (AvroRuntimeException e) {
+        LOG.error("Could not set Field [" + columnNames.get(i).name() + "] in the generic record");
+        fieldErrorCnt++;
+      } catch (Exception e) {
+        LOG.error("Parse Exception");
+        fieldErrorCnt++;
+      }
+    }
+
+    return rec;
+
+  }
+
+  @Override
+  public void teardown()
+  {
+    // TODO Auto-generated method stub
```

Comment:
Please remove this comment. I suggest to extend the class BaseOperator class instead of implementing Operator interface in which case you would not need to have any of these empty method.